### PR TITLE
add HRRR CONUS grid

### PR DESCRIPTION
**Pull request by simonbyrne**
Adds support for a Lambert Conformal Conic projection (used by the HRRR CONUS data)

cc @nbren12

### CODE REVIEW

File: earth2grid/lcc.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+from earth2grid import base
+from earth2grid._regrid import BilinearInterpolator
+
+try:
+    import pyvista as pv
+except ImportError:
+    pv = None
+
+
```
Comment (nbren12) [inline earth2grid/lcc.py:26]: Can you clarify the public API of this module like this
```
__all__ = [
    "LambertConformalConicGrid",
]
```

File: earth2grid/lcc.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            UN =   cos(theta) * U + sin(theta) * V
+            VN = - sin(theta) * U + cos(theta) * V
+        """
+        # center about reference longitude
+        delta_lon = lon - self.lon0
+        delta_lon = delta_lon - np.round(delta_lon/360) * 360 # convert to [-180, 180]
+        return self.n * np.deg2rad(delta_lon)
+
+
+    def proj(self, lat, lon):
```
Comment (nbren12) [inline earth2grid/lcc.py:79]: nit: let's avoid abbreviating short words
```suggestion
    def project(self, lat, lon):
```

File: earth2grid/lcc.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        Compute the projected x,y from lat,lon.
+        """
+        rho = self._rho(lat)
+        theta = self._theta(lon)
+    
+        x = rho * np.sin(theta)
+        y = self.rho0 - rho * np.cos(theta)
+        return x, y
+
+    def inv(self, x, y):
```
Comment (nbren12) [inline earth2grid/lcc.py:90]:
```suggestion
    def inverse_project(self, x, y):
```

File: earth2grid/lcc.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        lat = np.rad2deg(2 * np.arctan(np.power(self.RF/rho, 1/self.n))) - 90
+        lon = self.lon0 + np.rad2deg(theta / self.n)
+        return lat, lon
+
+# Projection used by HRRR CONUS (Continental US) data
+HRRR_CONUS_PROJECTION = LambertConformalConicProjection(
+    lon0 = -97.5,
+    lat0 = 38.5,
+    lat1 = 38.5,
+    lat2 = 38.5,
```
Comment (nbren12) [inline earth2grid/lcc.py:106]: Should these all be 38.5? just double checking.

File: earth2grid/lcc.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            raise ImportError("Need to install pyvista")
+
+        lat, lon = self.lat_lon
+        y = np.cos(np.deg2rad(lat)) * np.sin(np.deg2rad(lon))
+        x = np.cos(np.deg2rad(lat)) * np.cos(np.deg2rad(lon))
+        z = np.sin(np.deg2rad(lat))
+        grid = pv.StructuredGrid(x, y, z)
+        return grid
+
+def hrrr_conus_grid(ix0 = 0, iy0 = 0, nx = 1799, ny = 1059):
```
Comment (nbren12) [inline earth2grid/lcc.py:165]: linter will typically reject spaces before and after "=". Please run linter.

File: earth2grid/lcc.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+        self.shape = lat.shape
+        self._bilinear = BilinearInterpolator(
+            x_coords = torch.from_numpy(src.x), 
+            y_coords = torch.from_numpy(src.y), 
+            x_query = torch.from_numpy(x.ravel()),
+            y_query = torch.from_numpy(y.ravel()))
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        out = self._bilinear(x)
+        return out.view(out.shape[:-1] + self.shape)
```
Comment (nbren12) [inline earth2grid/lcc.py:200]: Is this wrapper class (_RegridFromLCC) necessary? I thought this reshaping behavior was already supported by BilinearInterpolator.

File: tests/test_lcc.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#%%
+from earth2grid.lcc import HRRR_CONUS_GRID
+import numpy as np
+import torch
+from pytest import approx
```
Comment (nbren12) [inline tests/test_lcc.py:20]: nit: For 3rd party packages, I roughly try to follow [Google's style guide for imports](https://google.github.io/styleguide/pyguide.html#22-imports) which forbids importing members from modules.


```suggestion
import pytest
```

and then `pytest.approx` below.

**Review by nbren12 — CHANGES_REQUESTED** [review]
Awesome! Some nits below, but please run the linter and fix any issues that come up.

File: earth2grid/lcc.py (hunk 4 above)
Comment (simonbyrne) [reply earth2grid/lcc.py:106]: Yes, see https://rapidrefresh.noaa.gov/hrrr/HRRR_conus.domain.txt

File: earth2grid/lcc.py (hunk 6 above)
Comment (simonbyrne) [reply earth2grid/lcc.py]: BilinearInterpolator wasn't working for multidimensional destination grids, but I think I've fixed that now.

**Comment by nbren12** [comment]
LGTM

**Pull request merged**